dll/win32/shell32/utils.cpp

```diff
@@ -62,6 +62,71 @@ EXTERN_C DWORD WINAPI SHGetUserSessionId(_In_opt_ HANDLE hToken)
     return dwSessionId;
 }
 
+// Helper class for SHInvokePrivilegedFunctionW
+struct CPrivilegeEnable
```

Comment:
I'm not sure why it is necessary to have a struct/class for that function. All the code used there (the constructor and destructor) could be instead placed inside the implementation of the SHInvokePrivilegedFunctionW.

In other words, this weird choice of implementation just looks suspicious (and you understand what I mean)

Reply:
Fixed.